sklearn/utils/_array_api.py

```diff
@@ -98,7 +98,7 @@ def _isdtype_single(dtype, kind, *, xp):
                 for k in ("signed integer", "unsigned integer")
             )
         elif kind == "real floating":
-            return dtype in {xp.float32, xp.float64}
+            return dtype in supported_float_dtypes(xp)
```

Comment:
Can we add a test for this and update add the type in`_NumPyAPIWrapper`:

```patch
diff --git a/sklearn/utils/_array_api.py b/sklearn/utils/_array_api.py
index 852713ce7f..6a7bacc6b7 100644
--- a/sklearn/utils/_array_api.py
+++ b/sklearn/utils/_array_api.py
@@ -226,6 +226,7 @@ class _NumPyAPIWrapper:
         "uint16",
         "uint32",
         "uint64",
+        "float16",
         "float32",
         "float64",
         "complex64",
diff --git a/sklearn/utils/tests/test_array_api.py b/sklearn/utils/tests/test_array_api.py
index 77fa20e6d0..841a274425 100644
--- a/sklearn/utils/tests/test_array_api.py
+++ b/sklearn/utils/tests/test_array_api.py
@@ -255,6 +255,9 @@ def test_get_namespace_array_api_isdtype(wrapper):
     assert xp.isdtype(xp.float64, "real floating")
     assert not xp.isdtype(xp.int32, "real floating")
 
+    if isinstance(xp, _NumPyAPIWrapper):
+        assert xp.isdtype(xp.float16, "real floating")
+
     assert xp.isdtype(xp.bool, "bool")
     assert not xp.isdtype(xp.float32, "bool")
```

Reply:
I did something slightly different for the test. Used `supported_float_dtypes` instead of testing `float16` only for Numpy.